src/widget/sidebar/city.c

```diff
@@ -293,22 +293,26 @@ static void button_build(int submenu, int param2)
 
 static void button_undo(int param1, int param2)
 {
+    window_build_menu_show(SUBMENU_NONE);
```

Comment:
Instead of "show none", it's more readable to create a new function `window_build_menu_hide()` that internally calls `window_build_menu_show(SUBMENU_NONE)`. That also gets rid of the SUBMENU_NONE define in `build_menu.h`